File: fc-chat/demo/part3/ch3/03-2-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/FetchConnectionsRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.constant.UserConnectionStatus;

public class FetchConnectionsRequest extends BaseRequest {

  private final UserConnectionStatus status;

  @JsonCreator
  public FetchConnectionsRequest(@JsonProperty("status") UserConnectionStatus status) {
    super(MessageType.FETCH_CONNECTIONS_REQUEST);
    this.status = status;
  }

  public UserConnectionStatus getStatus() {
    return status;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/FetchConnectionsRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.constant.UserConnectionStatus

class FetchConnectionsRequest @JsonCreator constructor(@JsonProperty("status") val status: UserConnectionStatus) :
    BaseRequest(MessageType.FETCH_CONNECTIONS_REQUEST) {
}
